### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by Laylo309**
- [x] Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- [x] Follow SOLID principles.
- [x] Pay special attention to the "Single-responsibility Principle".
    - Make sure that the role of `main.rb` is only to start your code and ask the user for the initial choice of options (or to quit).
    - Extract some pieces of logic into a separate class(es).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    puts 'Select a book from the following list by number'
+    @books.each_with_index do |book, index|
+      puts "#{index} Title: #{book.title}, Author: #{book.author}"
+    end
+
+    book_rental = gets.chomp.to_i
+
+    print "\nSelect a person from th following list by number (not id)"
+    @people.each_with_index do |person, index|
+      puts "#{index} #{person.class} Name: #{person.name}, Id: #{person.id}, Age: #{person.age}"
```
Comment (rloterh) [inline app.rb]: - [x] Kindly ensure that a registered person's detail fall on the next line, as can be seen from the [video example](https://www.youtube.com/watch?v=vkkgrhD6aXQ)

image

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
+require './person'
+require './book'
+require './classroom'
+require './rental'
+require './student'
+require './teacher'
+#rubocop:disable all
```
Comment (rloterh) [inline app.rb]: - [x] You have **disabled linter check** on your code. Please enable linter check on your code to ensure that your code is quality and clean.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Nelsino & Laylo, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍  
- Code follows standard practices
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (henatan99) [reply app.rb]: This issue has been addressed well. Good job! :clap: :clap:

**Review by henatan99 — APPROVED** [review]
Hi @Laylo309 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**